publisher-sdk/src/main/java/com/criteo/publisher/adview/MraidInteractor.kt

```diff
@@ -0,0 +1,51 @@
+/*
+ *    Copyright 2020 Criteo
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package com.criteo.publisher.adview
+
+import android.webkit.WebView
+import com.criteo.publisher.annotation.OpenForTesting
+import com.criteo.publisher.logging.LoggerFactory
+
+@OpenForTesting
+internal class MraidInteractor(private val webView: WebView) {
+
+  private val logger = LoggerFactory.getLogger(MraidInteractor::class.java)
+
+  fun notifyReady() {
+    callOnMraidObject("notifyReady()")
+  }
+
+  fun notifyError(message: String, action: String? = null) {
+    callOnMraidObject("notifyError(${asJsArgs(message, action)})")
+  }
+
+  private fun callOnMraidObject(jsCode: String) {
+    val jsToEvaluate = "window.mraid.$jsCode"
+    logger.debug("Calling mraid object with js: $jsToEvaluate")
+    webView.evaluateJavascript(jsToEvaluate, null)
+  }
+
+  private fun asJsArgs(vararg params: Any?): String {
+    return params.joinToString(separator = ", ") {
+      when (it) {
+        null -> "undefined"
+        is String -> "\"$it\""
+        else -> it.toString()
```

Comment:
I would rather throw in default case. Normally, if you need a new type, you should quickly detect it with the exception and fix it. It will prevent you to blindly guess that all types are supported while they are not

Reply:
Okay, it makes sense in a long run. Will do the change